apps/teamsjs-diagnostic-app/src/components/sample/ApiComponents.tsx

```diff
@@ -1,39 +1,389 @@
+import AppInstallDialogAPIs from './../../apis/AppInstallDialogApi';
+import BarCodeAPIs from './../../apis/BarCodeApi';
+import DialogAPIs from '../../apis/DialogApi';
+import ChatAPIs from '../../apis/ChatApi';
+import CallAPIs from '../../apis/CallApi';
+import CalendarAPIs from '../../apis/CalendarApi';
+import DialogCardAPIs from '../../apis/DialogCardApi';
+import PagesAPIs from '../../apis/PagesApi';
+import ProfileAPIs from '../../apis/ProfileApi';
+import SearchAPIs from '../../apis/SearchApi';
+import ClipboardAPIs from '../../apis/ClipboardApi';
+import GeolocationAPIs from '../../apis/GeolocationApi';
+import SharingAPIs from '../../apis/SharingApi';
+import StageViewAPIs from '../../apis/StageViewApi';
+import { stageView } from '@microsoft/teams-js';
+import PeopleAPIs from '../../apis/PeopleApi';
+import MenusAPIs from '../../apis/MenusApi';
+import PagesTabsAPIs from '../../apis/PagesTabsApi';
+import TeamsCoreAPIs from '../../apis/TeamsCoreApi';
+import SecondaryBrowserAPIs from '../../apis/SecondaryBrowserApi';
+
 export interface ApiComponent {
   title: string;
   name: string;
-  inputType: 'text' | 'checkbox' | 'none';
-  onClick: any;
+  functions: {
+    name: string;
+    requiresInput: boolean;
+  }[];
   defaultInput?: string;
-  defaultCheckboxState?: boolean;
-  label?: string;
-  options: string[];
+  onClick: any;
+  renderComponent?: (props: { apiComponent: ApiComponent; onDropToScenarioBox: (api: ApiComponent, func: string, input?: string) => void }) => JSX.Element;
 }
 
 const apiComponents: ApiComponent[] = [
   {
     title: 'App Install Dialog API',
     name: 'appInstallDialog',
-    options: ['OpenAppInstallDialog', 'CheckAppInstallCapability'],
-    defaultInput: 'default input for OpenAppInstallDialog',
-    inputType: 'text',
+    functions: [
+      { name: 'OpenAppInstallDialog', requiresInput: true },
+      { name: 'CheckAppInstallCapability', requiresInput: false },
+    ],
+    defaultInput: JSON.stringify({ appId: '957f8a7e-fbcd-411d-b69f-acb7eb58b515' }),
     onClick: () => console.log('App Install Dialog API called'),
+    renderComponent: (props) => <AppInstallDialogAPIs {...props} />
   },
   {
     title: 'Bar Code API',
     name: 'barCode',
-    options: ['checkBarCodeCapability', 'scanBarCode', 'hasBarCodePermission', 'requestBarCodePermission'],
+    functions: [
```

Comment:
Probably out of scope for your project given where we are relative to your last day and how soon that is, but it would really make this code a lot more maintainable if each JSON blob was exported by its corresponding *Api.tsx file and this big section of code here was able to dynamically crawl every component. Imagine a universe where adding a new component didn't mean a developer had to know to come here and update this list!

Anyways, I think the design change would be probably pretty big and out of scope, but I wanted to say it here anyways.